### DISCUSSION TITLE
Provide distroless images with sleep utility

### DISCUSSION THREAD

**Discussion by GeorgeTsilias** [istio/istio#48890] [Ideas]
Hello everyone,
How would you feel like providing a distroless image variant that includes the `sleep` utility, statically linked on the filesystem?
One other idea would be to implement `sleep` as a command on some of Istio's binaries, but I guess that would be more time consuming.

We're currently trying to rollout distroless istio-proxy images on our clusters in order to minimise the attack surface, but since we're using AWS NLBs with IP targets we need a preStop hook that makes the containers `sleep` for X number of seconds before SIGTERM is sent. This is due to a (widely known) limitation on the AWS side but I don't see anything being done to overcome it from their side, and I'm sure that there are a lot of people that are affected by this. 
 
More information on why the `sleep` preStop hook is needed can be found on the links below:
[1] https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/2366
[2] https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/2131

EDIT: Just going through some of the related Issues, it seems like there was another request like this one here: https://github.com/istio/istio/issues/47779

**Comment by howardjohn** [comment]
You should not need sleep, use terminationDrainDuration.

I sent a PR on this: https://github.com/istio/istio/pull/48996, but it was rejected
